spec/unit/complex_example_spec.rb

```diff
@@ -90,4 +90,12 @@
   it "should not be able to fire unknown events" do
     expect(auth.aasm.may_fire_event?(:unknown)).to be false
   end
+
+  it "should raise AASM::UndefinedState when firing unknown events" do
+    expect { auth.aasm.fire(:unknown) }.to raise_error(AASM::UndefinedState, "State :unknown doesn't exist")
+  end
+
+  it "should raise AASM::UndefinedState when firing unknown bang events" do
+    expect { auth.aasm.fire(:unknown!) }.to raise_error(AASM::UndefinedState, "State :unknown! doesn't exist")
+  end
```

Comment:
We should add specs to make sure that callbacks are not triggered when `UndefinedState` exception is raised

Reply:
Sounds good - I'll see if I can understand how callbacks get triggered, it will be fun 🙂